rest-fights/src/test/java/io/quarkus/sample/superheroes/fight/DisableIfContinuousTesting.java

```diff
@@ -0,0 +1,46 @@
+package io.quarkus.sample.superheroes.fight;
+
+import io.quarkus.sample.superheroes.fight.DisableIfContinuousTesting.ContinuousTestingCondition;
+
+import org.junit.jupiter.api.extension.ConditionEvaluationResult;
+import org.junit.jupiter.api.extension.ExecutionCondition;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.extension.ExtensionConfigurationException;
+import org.junit.jupiter.api.extension.ExtensionContext;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.TYPE)
+@ExtendWith(ContinuousTestingCondition.class)
+public @interface DisableIfContinuousTesting {
+
+    class ContinuousTestingCondition implements ExecutionCondition {
+      @Override
+      public ConditionEvaluationResult evaluateExecutionCondition(ExtensionContext context) {
+        final Class clazz = context.getRequiredTestClass();
+        final var annotation = clazz.getDeclaredAnnotation(DisableIfContinuousTesting.class);
+        if (annotation == null) {
+          throw new ExtensionConfigurationException("Could not find @" + DisableIfContinuousTesting.class + " annotation on the class " + clazz);
+        }
+
+        boolean isContinuousTesting = false;
```

Comment:
Couldn't this whole section be done a bit cleaner (& more Java >= 1.8 friendly)?

Some like

```java
var isContinuousTesting = Arrays.stream(new Exception().getStackTrace())
          .map(StackTraceElement::getClassName)
          .filter(className -> className.contains("io.quarkus.deployment.dev.testing"))
          .count() > 0;

        return isContinuousTesting ? ConditionEvaluationResult.disabled("launched in dev or test mode"): ConditionEvaluationResult.enabled("launched in normal mode");
```

Reply:
> Or just add the tag directly and avoid the need for the extra class.

I'm not really sure what that means, but sounds like the simplest idea and the easiest to maintain?

Reply:
Done and pushed. The meaning of the tag is slightly less obvious in the code than a distinct @DisableIfContinuousTesting annotation, but it gets rid of a class. If we were going to use it a lot I'd keep the annotation but just make it an alias for the `Tag`, but since we won't - hopefully - reduced clutter is probably more important.

I've confirmed that the normal mode consumer tests have the ability to fail, and dev mode tests pass.